From 36c9a7d2aa5b7013806a2fc6e216bf26225cfb97 Mon Sep 17 00:00:00 2001
From: Aravind Ramalingam <ramaliar@amazon.com>
Date: Wed, 15 Dec 2021 15:21:07 -0800
Subject: [PATCH] Modifying workflow action to image2disk

---
 internal/templates/templates.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/internal/templates/templates.go b/internal/templates/templates.go
index 1ec2e92..ac8b95b 100644
--- a/internal/templates/templates.go
+++ b/internal/templates/templates.go
@@ -67,7 +67,7 @@ tasks:
       - /lib/firmware:/lib/firmware:ro
     actions:
       - name: "stream-image"
-        image: oci2disk:v1.0.0
+        image: image2disk:v1.0.0
         timeout: 360
         environment:
           IMG_URL: %s
-- 
2.34.1

