From 4c30ba65baf17e89a7b4d6f0a5161157c30d07bf Mon Sep 17 00:00:00 2001
From: Aravind Ramalingam <ramaliar@amazon.com>
Date: Wed, 8 Jun 2022 10:11:11 -0700
Subject: [PATCH] Replace poweroff poweron with reset

Signed-off-by: Aravind Ramalingam <ramaliar@amazon.com>
---
 controllers/machine.go | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/controllers/machine.go b/controllers/machine.go
index 8ad050e..c670261 100644
--- a/controllers/machine.go
+++ b/controllers/machine.go
@@ -574,8 +574,7 @@ func (mrc *machineReconcileContext) getBMCJob(jName string, bmj *rufiov1.BMCJob)
 
 // createBMCJob creates a BMCJob object with the required tasks for hardware provisioning.
 func (mrc *machineReconcileContext) createBMCJob(hardware *tinkv1.Hardware, name string) error {
-	powerOffAction := rufiov1.HardPowerOff
-	powerOnAction := rufiov1.PowerOn
+	powerResetAction := rufiov1.Reset
 	controller := true
 	bmcJob := &rufiov1.BMCJob{
 		ObjectMeta: metav1.ObjectMeta{
@@ -597,9 +596,6 @@ func (mrc *machineReconcileContext) createBMCJob(hardware *tinkv1.Hardware, name
 				Namespace: mrc.tinkerbellMachine.Namespace,
 			},
 			Tasks: []rufiov1.Task{
-				{
-					PowerAction: &powerOffAction,
-				},
 				{
 					OneTimeBootDeviceAction: &rufiov1.OneTimeBootDeviceAction{
 						Devices: []rufiov1.BootDevice{
@@ -609,7 +605,7 @@ func (mrc *machineReconcileContext) createBMCJob(hardware *tinkv1.Hardware, name
 					},
 				},
 				{
-					PowerAction: &powerOnAction,
+					PowerAction: &powerResetAction,
 				},
 			},
 		},
-- 
2.34.1

