From 6e84664577272b2a1d941d3392cde725d9570636 Mon Sep 17 00:00:00 2001
From: Abhinav Pandey <abhinavmpandey08@gmail.com>
Date: Mon, 14 Feb 2022 16:06:44 -0800
Subject: [PATCH] Remove requirement for machine.Spec.Version

Signed-off-by: Abhinav Pandey <abhinavmpandey08@gmail.com>
---
 controllers/base.go | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/controllers/base.go b/controllers/base.go
index bfb28a2..9b84c3e 100644
--- a/controllers/base.go
+++ b/controllers/base.go
@@ -338,12 +338,6 @@ func isMachineReady(machine *clusterv1.Machine) (string, error) {
 		return "retrieving bootstrap data: linked Machine's bootstrap.dataSecretName is not available yet", nil
 	}
 
-	// Spec says this field is optional, but @detiber says it's effectively required,
-	// so treat it as so.
-	if machine.Spec.Version == nil || *machine.Spec.Version == "" {
-		return "", ErrMachineVersionEmpty
-	}
-
 	return "", nil
 }
 
-- 
2.32.0 (Apple Git-132)

