From eaa5811ede0e998454fd3b8be2c1813f9082ad7d Mon Sep 17 00:00:00 2001
From: Aravind Ramalingam <ramaliar@amazon.com>
Date: Fri, 11 Mar 2022 10:53:25 -0800
Subject: [PATCH] Temporary comment error return

---
 pbnj/controllers/controller.go | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/pbnj/controllers/controller.go b/pbnj/controllers/controller.go
index 591b55f..4ecbfad 100644
--- a/pbnj/controllers/controller.go
+++ b/pbnj/controllers/controller.go
@@ -75,7 +75,9 @@ func (r *Reconciler) reconcileNormal(ctx context.Context, bmc *pbnjv1alpha1.BMC)
 		if err != nil {
 			logger.Error(err, "Failed to set boot device", "BootDevice", bmc.Spec.BootDevice)
 
-			return ctrl.Result{}, fmt.Errorf("failed to set boot device: %s", bmc.Spec.BootDevice) //nolint:goerr113
+			// TODO: Temporary fix to not error out on boot set failure.
+			// This change will be reverted when we have better error propagations
+			// return ctrl.Result{}, fmt.Errorf("failed to set boot device: %s", bmc.Spec.BootDevice) //nolint:goerr113
 		}
 	}
 
-- 
2.34.1

