From 2dec4b6603c11128881825ae6bc314c09caa019b Mon Sep 17 00:00:00 2001
From: Amelia Lu <63176721+peirulu@users.noreply.github.com>
Date: Wed, 7 May 2025 16:47:06 -0700
Subject: [PATCH] upgrade go to 1.24.2

---
 go.mod | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/go.mod b/go.mod
index 6e83475..ec4b195 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/aquasecurity/harbor-scanner-trivy
 
-go 1.22
+go 1.24.2
 
 require (
 	github.com/alicebob/miniredis/v2 v2.33.0
-- 
2.48.1

