From 063343ddb028d7568736d89e3fb7354aaec33fbf Mon Sep 17 00:00:00 2001
From: "Ostosh, Ivy" <ivyjin@amazon.com>
Date: Fri, 13 Jan 2023 08:45:08 -0600
Subject: [PATCH] Update go.mod

---
 go.mod | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/go.mod b/go.mod
index 435e95b..4d4dc26 100644
--- a/go.mod
+++ b/go.mod
@@ -25,7 +25,7 @@ require (
 	github.com/prometheus/common v0.37.0
 	github.com/prometheus/exporter-toolkit v0.8.2
 	github.com/prometheus/procfs v0.8.0
-	github.com/safchain/ethtool v0.2.0
+	github.com/safchain/ethtool v0.1.1-0.20220305102649-532e6d132c2e
 	github.com/soundcloud/go-runit v0.0.0-20150630195641-06ad41a06c4a
 	golang.org/x/sys v0.2.0
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6
-- 
2.31.0

