From c5349f4e35a758687549bef6c6daf6b5438ecf36 Mon Sep 17 00:00:00 2001
From: jonahjon <jonahjones094@gmail.com>
Date: Mon, 21 Nov 2022 14:16:21 -0600
Subject: [PATCH] remove_go_version_lock

Signed-off-by: jonahjon <jonahjones094@gmail.com>
---
 build-aux/prelude_go.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build-aux/prelude_go.mk b/build-aux/prelude_go.mk
index 9547b2ca7..7de3d3244 100644
--- a/build-aux/prelude_go.mk
+++ b/build-aux/prelude_go.mk
@@ -75,7 +75,7 @@ _prelude.go.VERSION.ge = $(call _prelude.go.VERSION._ge,$(call _prelude.go.VERSI
 #
 # Evaluates to $(TRUE) if `go` is >= the specified version, $(FALSE)
 # otherwise.
-_prelude.go.VERSION.HAVE = $(if $(_prelude.go.HAVE),$(call _prelude.go.VERSION.ge,$(_prelude.go.VERSION),$1))
+_prelude.go.VERSION.HAVE = $(TRUE)
 
 #
 # Building Go programs for use by build-aux
-- 
2.34.1

