From 243aed81c050b3898927d03079282a06c215a42e Mon Sep 17 00:00:00 2001
From: Amelia Lu <peirulu@amazon.com>
Date: Thu, 5 Mar 2026 06:54:10 +0000
Subject: [PATCH] update go version to fix cve

---
 go.mod | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/go.mod b/go.mod
index b50f2bff0..af960fa42 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/aws-observability/aws-otel-collector
 
-go 1.24.11
+go 1.24.13
 
 require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/confmap/provider/s3provider v0.143.0
-- 
2.34.1

