From 7f3498373eb1d0cdf7e07e87bd065742655c6083 Mon Sep 17 00:00:00 2001
From: Vivek Koppuru <koppv@amazon.com>
Date: Wed, 12 Jan 2022 19:04:15 -0800
Subject: [PATCH] Add status.version to list of fields to ignore for update

---
 controlplane/kubeadm/internal/webhooks/kubeadmcontrolplane.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/controlplane/kubeadm/internal/webhooks/kubeadmcontrolplane.go b/controlplane/kubeadm/internal/webhooks/kubeadmcontrolplane.go
index 8d44b1839..b8120dee6 100644
--- a/controlplane/kubeadm/internal/webhooks/kubeadmcontrolplane.go
+++ b/controlplane/kubeadm/internal/webhooks/kubeadmcontrolplane.go
@@ -91,6 +91,7 @@ func (webhook *KubeadmControlPlane) ValidateCreate(_ context.Context, k *control
 
 const (
 	spec                 = "spec"
+	status               = "status"
 	kubeadmConfigSpec    = "kubeadmConfigSpec"
 	clusterConfiguration = "clusterConfiguration"
 	initConfiguration    = "initConfiguration"
@@ -198,6 +199,7 @@ func (webhook *KubeadmControlPlane) ValidateUpdate(_ context.Context, oldK, newK
 		{spec, "machineNaming", "*"},
 		{spec, "rollout"},
 		{spec, "rollout", "*"},
+		{status, "version"},
 	}
 
 	allErrs := validateKubeadmControlPlaneSpec(newK.Spec, field.NewPath("spec"))
-- 
2.48.1

