From c31cd2fca2460e2b1703f42003e96c10f0dff126 Mon Sep 17 00:00:00 2001
From: Vignesh Goutham Ganesh <vgg@amazon.com>
Date: Tue, 17 Aug 2021 17:09:03 -0700
Subject: [PATCH 10/36] Add network settings only when there is a proxy defined

---
 bootstrap/kubeadm/internal/bottlerocket/bootstrap.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bootstrap/kubeadm/internal/bottlerocket/bootstrap.go b/bootstrap/kubeadm/internal/bottlerocket/bootstrap.go
index 4ce098a4c..e4be8d945 100644
--- a/bootstrap/kubeadm/internal/bottlerocket/bootstrap.go
+++ b/bootstrap/kubeadm/internal/bottlerocket/bootstrap.go
@@ -38,6 +38,8 @@ no-proxy = "{{.NoProxyEndpoints}}"
 
 {{template "kubernetesInitSettings" .}}
 
+{{- if (ne .HTTPSProxyEndpoint "")}}
 {{template "networkInitSettings" .}}
+{{- end -}}
 `
 )
-- 
2.30.1

