From 162a773c79192c4b424932c40ae408d0e1e2d339 Mon Sep 17 00:00:00 2001
From: Vignesh Goutham Ganesh <vgg@amazon.com>
Date: Tue, 17 Aug 2021 17:09:03 -0700
Subject: [PATCH] Add network settings only when there is a proxy defined

Signed-off-by: Vignesh Goutham Ganesh <vgg@amazon.com>
---
 bootstrap/kubeadm/internal/bottlerocket/bootstrap.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bootstrap/kubeadm/internal/bottlerocket/bootstrap.go b/bootstrap/kubeadm/internal/bottlerocket/bootstrap.go
index 4ce098a4c..e4be8d945 100644
--- a/bootstrap/kubeadm/internal/bottlerocket/bootstrap.go
+++ b/bootstrap/kubeadm/internal/bottlerocket/bootstrap.go
@@ -38,6 +38,8 @@ no-proxy = "{{.NoProxyEndpoints}}"
 
 {{template "kubernetesInitSettings" .}}
 
+{{- if (ne .HTTPSProxyEndpoint "")}}
 {{template "networkInitSettings" .}}
+{{- end -}}
 `
 )
-- 
2.24.3 (Apple Git-128)

