From 2d12a7941bf21bc261917e85d8754128e9c686ef Mon Sep 17 00:00:00 2001
From: Rajashree Mandaogane <mandaor@amazon.com>
Date: Wed, 17 Nov 2021 10:44:42 -0800
Subject: [PATCH 28/41] Update log message based on 1.0.1 tag

---
 .../docker/controllers/dockermachine_controller.go            | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/infrastructure/docker/controllers/dockermachine_controller.go b/test/infrastructure/docker/controllers/dockermachine_controller.go
index b2f1255f9..abf54343f 100644
--- a/test/infrastructure/docker/controllers/dockermachine_controller.go
+++ b/test/infrastructure/docker/controllers/dockermachine_controller.go
@@ -196,7 +196,7 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 		// set address in machine status
 		machineAddress, err := externalMachine.Address(ctx)
 		if err != nil {
-			r.Log.Error(err, "failed to get the machine address")
+			log.Error(err, "failed to get the machine address")
 			return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
 		}
 
@@ -337,7 +337,7 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 			if errors.As(err, &docker.ContainerNotRunningError{}) {
 				return ctrl.Result{}, errors.Wrap(err, "failed to patch the Kubernetes node with the machine providerID")
 			}
-			r.Log.Error(err, "failed to patch the Kubernetes node with the machine providerID")
+			log.Error(err, "failed to patch the Kubernetes node with the machine providerID")
 			return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
 		}
 	}
-- 
2.27.0

