From 57c79b439f61b01ff78ab50909ade8be5bdd17b5 Mon Sep 17 00:00:00 2001
From: Jackson West <jgw@amazon.com>
Date: Sat, 19 Aug 2023 09:35:39 -0500
Subject: [PATCH 28/39] disable cgroupns=private to fix AL2

---
 test/infrastructure/container/docker.go | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/test/infrastructure/container/docker.go b/test/infrastructure/container/docker.go
index a3a43945e..f7c8bc7f5 100644
--- a/test/infrastructure/container/docker.go
+++ b/test/infrastructure/container/docker.go
@@ -41,8 +41,6 @@ import (
 	"github.com/pkg/errors"
 	kerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/utils/ptr"
-
-	"sigs.k8s.io/cluster-api/test/infrastructure/kind"
 )
 
 const (
@@ -411,10 +409,11 @@ func (d *dockerRuntime) RunContainer(ctx context.Context, runConfig *RunContaine
 	}
 	networkConfig := network.NetworkingConfig{}
 
-	// NOTE: starting from Kind 0.20 kind requires CgroupnsMode to be set to private.
-	if runConfig.KindMode != kind.ModeNone && runConfig.KindMode != kind.Mode0_19 {
-		hostConfig.CgroupnsMode = "private"
-	}
+	// AWS: groupns = private breaks on AL2 nodes, kind 0.20 still "supports" non-private mode
+	// but it is deprecated it. For now we revert to the previous behavior.
+	// if runConfig.KindMode != kind.ModeNone && runConfig.KindMode != kind.Mode0_19 {
+	// 	hostConfig.CgroupnsMode = "private"
+	// }
 
 	if runConfig.IPFamily == IPv6IPFamily || runConfig.IPFamily == DualStackIPFamily {
 		hostConfig.Sysctls = map[string]string{
-- 
2.48.1

