From b8f110820a5f503f3a948c92d06df77f1bf678d7 Mon Sep 17 00:00:00 2001
From: Prow Bot <prow@amazonaws.com>
Date: Wed, 23 Oct 2024 13:02:26 -0700
Subject: [PATCH 2/2] Add latest CAPC minor release v0.5.0 to metadata.yaml

---
 metadata.yaml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/metadata.yaml b/metadata.yaml
index bd61395..836f295 100644
--- a/metadata.yaml
+++ b/metadata.yaml
@@ -9,3 +9,6 @@ releaseSeries:
   - major: 0
     minor: 4
     contract: v1beta1
+  - major: 0
+    minor: 5
+    contract: v1beta1
-- 
2.47.0

