From ed0333cad31eacd8d597fd0e49d4b3d79e8ab9fb Mon Sep 17 00:00:00 2001
From: Prow Bot <prow@amazonaws.com>
Date: Thu, 5 Mar 2026 07:21:04 +0000
Subject: [PATCH] update go version to fix cve

---
 go.mod | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/go.mod b/go.mod
index 6f9a9a3c..2fa68adc 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module sigs.k8s.io/metrics-server
 
-go 1.24.12
+go 1.24.13
 
 require (
 	github.com/google/go-cmp v0.7.0
-- 
2.34.1

