From 1d7b9719b3716285fa0d6a876141447d85c52ac5 Mon Sep 17 00:00:00 2001
From: Abhay Krishna Arunachalam <arnchlm@amazon.com>
Date: Wed, 15 May 2024 14:14:38 -0400
Subject: [PATCH] Revert zero value change for capacityInKB

---
 govc/vm/disk/change.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/govc/vm/disk/change.go b/govc/vm/disk/change.go
index d57b1bd4..1643a917 100644
--- a/govc/vm/disk/change.go
+++ b/govc/vm/disk/change.go
@@ -154,7 +154,7 @@ func (cmd *change) Run(ctx context.Context, f *flag.FlagSet) error {
 
 	if int64(cmd.bytes) != 0 {
 		editdisk.CapacityInBytes = int64(cmd.bytes)
-		editdisk.CapacityInKB = int64(0) // zero deprecated field
+		editdisk.CapacityInKB = int64(cmd.bytes) / 1024
 	}
 
 	if editdisk.StorageIOAllocation == nil {
-- 
2.39.3 (Apple Git-146)

