From fd6dfbad1f90c8440068757b09542f3876b8ac20 Mon Sep 17 00:00:00 2001
From: Rajashree Mandaogane <mandaor@amazon.com>
Date: Thu, 3 Mar 2022 14:58:30 -0800
Subject: [PATCH 3/4] Increase wait backoff retries during join

---
 constants/constants.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/constants/constants.go b/constants/constants.go
index e5e421ac..82bb4221 100644
--- a/constants/constants.go
+++ b/constants/constants.go
@@ -162,7 +162,7 @@ fi
 "{{ .EtcdctlExecutable }}" "$@"
 `
 
-	DefaultBackOffSteps    = 5
+	DefaultBackOffSteps    = 25
 	DefaultBackOffDuration = 2 * time.Second
 	DefaultBackOffFactor   = 2.0
 )
-- 
2.39.2

