From 567a7db4a0e6c1eb3535f3d9c7b7644b9da50dbd Mon Sep 17 00:00:00 2001
From: Guillermo Gaston <gaslor@amazon.com>
Date: Sat, 7 Aug 2021 10:04:32 +0000
Subject: [PATCH 9/9] Fixed linting

---
 constants/constants.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/constants/constants.go b/constants/constants.go
index bcee405c..e5f6be80 100644
--- a/constants/constants.go
+++ b/constants/constants.go
@@ -33,7 +33,7 @@ const (
 	EnvironmentFile  = "/etc/etcd/etcd.env"
 	EtcdctlEnvFile   = "/etc/etcd/etcdctl.env"
 
-	DefaultDataDir = "/var/lib/etcd"
+	DefaultDataDir    = "/var/lib/etcd"
 	DefaultPodSpecDir = "/etc/kubernetes/manifests"
 
 	DefaultLoopbackHost = "127.0.0.1"
-- 
2.25.1

