From 4551757dd574cd87f37201ab55e116d9a643cee5 Mon Sep 17 00:00:00 2001
From: pytimer <lixin20101023@gmail.com>
Date: Sat, 19 Jun 2021 21:50:51 +0800
Subject: [PATCH] Support linux other arch

---
 binary/etcd.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/binary/etcd.go b/binary/etcd.go
index 4427ff93..b6cc9de3 100644
--- a/binary/etcd.go
+++ b/binary/etcd.go
@@ -22,6 +22,7 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"runtime"
 	"strings"
 	"time"
 
@@ -110,7 +111,7 @@ func Download(releaseURL, version, locationDir string) error {
 }
 
 func releaseFile(version string) string {
-	return fmt.Sprintf("etcd-v%s-linux-amd64.tar.gz", version)
+	return fmt.Sprintf("etcd-v%s-linux-%s.tar.gz", version, runtime.GOARCH)
 }
 
 func downloadURL(releaseURL, version string) string {
-- 
2.33.1

