From a39a7c88f9bb36a775c1d484c2f729e37d20a07b Mon Sep 17 00:00:00 2001
From: Rajashree Mandaogane <mandaor@amazon.com>
Date: Thu, 3 Mar 2022 14:58:30 -0800
Subject: [PATCH] Increase wait backoff retries during join

---
 constants/constants.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/constants/constants.go b/constants/constants.go
index cc8daca1..95959325 100644
--- a/constants/constants.go
+++ b/constants/constants.go
@@ -162,7 +162,7 @@ fi
 "{{ .EtcdctlExecutable }}" "$@"
 `
 
-	DefaultBackOffSteps    = 5
+	DefaultBackOffSteps    = 25
 	DefaultBackOffDuration = 2 * time.Second
 	DefaultBackOffFactor   = 2.0
 )
-- 
2.30.1

