From 3956d6443c5fffed03e40a5dae6da7102bcab1ef Mon Sep 17 00:00:00 2001
From: Prow Bot <prow@amazonaws.com>
Date: Mon, 7 Oct 2024 19:35:59 +0000
Subject: [PATCH 4/4] pass cpu host when running qemu to support testing al23

---
 src/cmd/linuxkit/run_qemu.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/cmd/linuxkit/run_qemu.go b/src/cmd/linuxkit/run_qemu.go
index ecf95ee67..d6df98d57 100644
--- a/src/cmd/linuxkit/run_qemu.go
+++ b/src/cmd/linuxkit/run_qemu.go
@@ -466,6 +466,9 @@ func buildQemuCmdline(config QemuConfig) (QemuConfig, []string) {
 		} else {
 			qemuArgs = append(qemuArgs, "-cpu", "cortex-a57")
 		}
+	} else {
+		// aws - We require x86_64 to run al23 with its version of glibc
+		qemuArgs = append(qemuArgs, "-cpu", "host")
 	}
 
 	// goArch is the GOARCH equivalent of config.Arch
-- 
2.49.0

