From f4e900868e0e2fdd55c8c1176ab4e3629e5e5fd2 Mon Sep 17 00:00:00 2001
From: Aravind Ramalingam <ramaliar@amazon.com>
Date: Fri, 11 Feb 2022 12:24:32 -0800
Subject: [PATCH] Modifying docker path for AL2 base image

---
 tink-docker/main.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tink-docker/main.go b/tink-docker/main.go
index 6e96694..40650ee 100644
--- a/tink-docker/main.go
+++ b/tink-docker/main.go
@@ -46,7 +46,7 @@ func main() {
 	fmt.Println("Downloaded the repository certificates, starting the Docker Engine")
 
 	// Build the command, and execute
-	cmd := exec.Command("/usr/local/bin/docker-init", "/usr/local/bin/dockerd")
+	cmd := exec.Command("/usr/bin/docker-init", "/usr/bin/dockerd")
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	err = cmd.Run()
-- 
2.34.1

