From 4b3830fb37adb5d70912c83c65e59ad256c2ad25 Mon Sep 17 00:00:00 2001
From: Vignesh Goutham Ganesh <vgg@amazon.com>
Date: Wed, 20 Sep 2023 10:33:44 -0500
Subject: [PATCH 9/10] Default Flatcar version to avoid pulling from internet on
 every make

Signed-off-by: Vignesh Goutham Ganesh <vgg@amazon.com>
---
 images/capi/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/images/capi/Makefile b/images/capi/Makefile
index f841eac48..905407368 100644
--- a/images/capi/Makefile
+++ b/images/capi/Makefile
@@ -312,7 +312,7 @@ WINDOWS_VERSIONS		:=	windows-2019 windows-2022
 
 # Set Flatcar Container Linux channel and version if not supplied
 FLATCAR_CHANNEL ?= stable
-FLATCAR_VERSION ?= current
+FLATCAR_VERSION ?= 3510.2.7
 ifeq ($(FLATCAR_VERSION),current)
 override FLATCAR_VERSION := $(shell hack/image-grok-latest-flatcar-version.sh $(FLATCAR_CHANNEL))
 endif
-- 
2.34.1

