From b6c334e70a201503f4a4995e505a75ab92fc7a82 Mon Sep 17 00:00:00 2001
From: Abhay Krishna Arunachalam <arnchlm@amazon.com>
Date: Mon, 13 Sep 2021 20:03:18 -0700
Subject: [PATCH] Replace promptui mod to mitigate LGPL dependency

---
 go.mod | 1 +
 1 file changed, 1 insertion(+)

diff --git a/go.mod b/go.mod
index 9f98c8c..9046b26 100644
--- a/go.mod
+++ b/go.mod
@@ -76,4 +76,5 @@ replace (
 	k8s.io/metrics => k8s.io/metrics v0.20.2
 	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.20.2
 	sigs.k8s.io/controller-runtime => github.com/kubernetes-sigs/controller-runtime v0.8.3
+	github.com/manifoldco/promptui => github.com/nguyer/promptui v0.8.1-0.20210517132806-70ccd4709797
 )
-- 
2.33.0

