From 007609565a99d96cf7a002a9fec73970fa6350c6 Mon Sep 17 00:00:00 2001
From: Yue Tan <yuetant@amazon.com>
Date: Tue, 2 Aug 2022 16:19:45 -0700
Subject: [PATCH] Fix saving logs

---
 pkg/collect/logs.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pkg/collect/logs.go b/pkg/collect/logs.go
index 8e33d6b6..62c26147 100644
--- a/pkg/collect/logs.go
+++ b/pkg/collect/logs.go
@@ -52,7 +52,7 @@ func Logs(c *Collector, logsCollector *troubleshootv1beta2.Logs) (CollectorResul
 						if containerName != "" {
 							key = fmt.Sprintf("%s/%s/%s-errors.json", logsCollector.Name, pod.Name, containerName)
 						}
-						output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
+						err := output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
 						if err != nil {
 							return nil, err
 						}
@@ -67,7 +67,7 @@ func Logs(c *Collector, logsCollector *troubleshootv1beta2.Logs) (CollectorResul
 					containerLogs, err := savePodLogs(ctx, c.BundlePath, client, pod, logsCollector.Name, container, logsCollector.Limits, false)
 					if err != nil {
 						key := fmt.Sprintf("%s/%s/%s-errors.json", logsCollector.Name, pod.Name, container)
-						output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
+						err := output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
 						if err != nil {
 							return nil, err
 						}
