From 7c2e46274a80f39f5886c9ec3385758e18180764 Mon Sep 17 00:00:00 2001
From: Ahree Hong <ahreeh@amazon.com>
Date: Fri, 29 Mar 2024 18:34:15 -0700
Subject: [PATCH 2/2] Set proxy in registry client

Signed-off-by: Ahree Hong <ahreeh@amazon.com>
---
 pkg/registry/util.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkg/registry/util.go b/pkg/registry/util.go
index 8baf0852..2b0dab0c 100644
--- a/pkg/registry/util.go
+++ b/pkg/registry/util.go
@@ -156,6 +156,7 @@ func NewRegistryClientWithTLS(out io.Writer, certFile, keyFile, caFile string, i
 		ClientOptHTTPClient(&http.Client{
 			Transport: &http.Transport{
 				TLSClientConfig: tlsConf,
+				Proxy:           http.ProxyFromEnvironment,
 			},
 		}),
 	)
-- 
2.40.0

