From 702d74591628e058871c9e63436187b4623f02dc Mon Sep 17 00:00:00 2001
From: Pooja Trivedi <tripooja@amazon.com>
Date: Thu, 4 Aug 2022 12:28:51 -0700
Subject: [PATCH] Set proxy environment in HTTP client

---
 pkg/registry/client.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkg/registry/client.go b/pkg/registry/client.go
index 4fb55020..6acf27e7 100644
--- a/pkg/registry/client.go
+++ b/pkg/registry/client.go
@@ -178,6 +178,7 @@ func (c *Client) newResolver(insecure, plainHTTP bool) (remotes.Resolver, error)
 			TLSClientConfig: &tls.Config{
 				InsecureSkipVerify: true,
 			},
+			Proxy:		    http.ProxyFromEnvironment,
 		}
 		opts = append(opts, auth.WithResolverClient(httpClient))
 	}
-- 
2.33.0

