From a641de95394b104f4dc5b79f60522ee69230296e Mon Sep 17 00:00:00 2001
From: Pooja Trivedi <tripooja@amazon.com>
Date: Thu, 4 Aug 2022 12:28:51 -0700
Subject: [PATCH 3/3] Set proxy environment in HTTP client

---
 pkg/registry/client.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkg/registry/client.go b/pkg/registry/client.go
index 3a5ff75a..1afcdd0e 100644
--- a/pkg/registry/client.go
+++ b/pkg/registry/client.go
@@ -189,6 +189,7 @@ func (c *Client) newResolver(insecure, plainHTTP bool) (remotes.Resolver, error)
 			TLSClientConfig: &tls.Config{
 				InsecureSkipVerify: true,
 			},
+			Proxy: http.ProxyFromEnvironment,
 		}
 		opts = append(opts, auth.WithResolverClient(httpClient))
 	}
-- 
2.39.2

