From b981945820e46efbb17f6ea00d9e8d31c9c01651 Mon Sep 17 00:00:00 2001
From: Vincent Ni <kangqini@amazon.com>
Date: Thu, 20 Jan 2022 10:57:50 -0800
Subject: [PATCH] remove base image check for cert installation

---
 make/photon/common/install_cert.sh | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/make/photon/common/install_cert.sh b/make/photon/common/install_cert.sh
index c536b5665..1a3811986 100755
--- a/make/photon/common/install_cert.sh
+++ b/make/photon/common/install_cert.sh
@@ -2,10 +2,6 @@
 
 set -e
 
-if ! grep -q "Photon" /etc/lsb-release; then
-    echo "Current OS is not Photon, skip appending ca bundle"
-    exit 0
-fi
 
 ORIGINAL_LOCATION=$(dirname "$0")
 
-- 
2.32.0 (Apple Git-132)

