From 4c34867af9c072b2d065f3f8976bf2e86609644f Mon Sep 17 00:00:00 2001
From: Vincent Ni <kangqini@amazon.com>
Date: Mon, 7 Feb 2022 17:41:55 -0800
Subject: [PATCH 1/2] remove base image check for cert installation

---
 make/photon/common/install_cert.sh | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/make/photon/common/install_cert.sh b/make/photon/common/install_cert.sh
index c536b5665..579708dd5 100755
--- a/make/photon/common/install_cert.sh
+++ b/make/photon/common/install_cert.sh
@@ -2,11 +2,6 @@
 
 set -e
 
-if ! grep -q "Photon" /etc/lsb-release; then
-    echo "Current OS is not Photon, skip appending ca bundle"
-    exit 0
-fi
-
 ORIGINAL_LOCATION=$(dirname "$0")
 
 if [ ! -f $ORIGINAL_LOCATION/ca-bundle.crt.original ]; then
-- 
2.32.0 (Apple Git-132)

