From 0a3185d132e4a4a43f11e5060240fff02fa966dd Mon Sep 17 00:00:00 2001
From: Prow Bot <prow@amazonaws.com>
Date: Tue, 29 Apr 2025 11:30:24 -0700
Subject: [PATCH] change the source of distribution for harbor makefile

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 2bbdcfe86..f16cfa176 100644
--- a/Makefile
+++ b/Makefile
@@ -110,9 +110,9 @@ TRIVYADAPTERVERSION=v0.33.0-rc.2
 NODEBUILDIMAGE=node:16.18.0
 
 # version of registry for pulling the source code
-REGISTRY_SRC_TAG=release/2.8
+REGISTRY_SRC_TAG=v3.0.0
 # source of upstream distribution code
-DISTRIBUTION_SRC=https://github.com/goharbor/distribution.git
+DISTRIBUTION_SRC=https://github.com/distribution/distribution.git
 
 # dependency binaries
 REGISTRYURL=https://storage.googleapis.com/harbor-builds/bin/registry/release-${REGISTRYVERSION}/registry
-- 
2.48.1

