From 5263d0e4df35768efceb4953a27844ac8c1a5f63 Mon Sep 17 00:00:00 2001
From: Prow Bot <prow@amazonaws.com>
Date: Thu, 17 Mar 2022 09:08:52 -0500
Subject: [PATCH 2/3] set user-agent for vsphere api calls

Signed-off-by: Prow Bot <prow@amazonaws.com>
---
 pkg/session/session.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkg/session/session.go b/pkg/session/session.go
index 0dd5860e..e489433d 100644
--- a/pkg/session/session.go
+++ b/pkg/session/session.go
@@ -187,6 +187,7 @@ func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *
 	if err != nil {
 		return nil, err
 	}
+	
 	vimClient.UserAgent = "k8s-capv-useragent"
 
 	c := &govmomi.Client{
-- 
2.39.2

