From 15575f1e76c8ec8172c15f558d2bd9e4dc9e6ab8 Mon Sep 17 00:00:00 2001
From: Prow Bot <prow@amazonaws.com>
Date: Thu, 17 Mar 2022 09:08:52 -0500
Subject: [PATCH] set user-agent for vsphere api calls

Signed-off-by: Prow Bot <prow@amazonaws.com>
---
 pkg/session/session.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pkg/session/session.go b/pkg/session/session.go
index 28ee585c..073c0ff9 100644
--- a/pkg/session/session.go
+++ b/pkg/session/session.go
@@ -177,6 +177,8 @@ func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *
 	if err != nil {
 		return nil, err
 	}
+	
+	vimClient.UserAgent = "k8s-capv-useragent"
 
 	c := &govmomi.Client{
 		Client:         vimClient,
-- 
2.35.1

