From 2972a803251a18e87ce92676cbac0aa079b32017 Mon Sep 17 00:00:00 2001
From: Abhinav Pandey <abhinavmpandey08@gmail.com>
Date: Tue, 3 May 2022 11:38:23 -0700
Subject: [PATCH] Remove .ONESHELL from rules.mk

Signed-off-by: Abhinav Pandey <abhinavmpandey08@gmail.com>
---
 rules.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/rules.mk b/rules.mk
index a881c42..027f521 100644
--- a/rules.mk
+++ b/rules.mk
@@ -5,8 +5,6 @@ MAKEFLAGS += --no-builtin-rules
 # This avoids a failure to then skip building on next run if the output is created by shell redirection for example
 # Not really necessary for now, but just good to have already if it becomes necessary later.
 .DELETE_ON_ERROR:
-# Treat the whole recipe as a one shell script/invocation instead of one-per-line
-.ONESHELL:
 # Use bash instead of plain sh
 SHELL := bash
 .SHELLFLAGS := -o pipefail -euc
-- 
2.30.1 (Apple Git-130)

