From c427b8fdd5551ee621867414fbfb3b73feb22e12 Mon Sep 17 00:00:00 2001
From: Aravind Ramalingam <ramaliar@amazon.com>
Date: Thu, 9 Dec 2021 17:40:32 -0800
Subject: [PATCH] Adding +build comment for go1.16 go-licenses compatability

---
 tools.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools.go b/tools.go
index 177df05..5091dcb 100644
--- a/tools.go
+++ b/tools.go
@@ -1,4 +1,5 @@
 //go:build tools
+// +build
 
 package tools
 
-- 
2.30.1

