From 703496132e2cb010225c2af077f7f558948f4622 Mon Sep 17 00:00:00 2001
From: Amelia Lu <63176721+peirulu@users.noreply.github.com>
Date: Mon, 15 Jun 2026 11:55:33 -0700
Subject: [PATCH 4/6] TEMP: lock containerd and runc version

---
 images/base/Dockerfile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/images/base/Dockerfile b/images/base/Dockerfile
index e7e73fe..f69c2cb 100644
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -93,6 +93,10 @@ RUN chmod 755 /kind /kind/bin && \
     # leaving for now, but al23 may not be affected by this issue
     && systemctl mask getty@tty1.service
 
+# see https://github.com/aws/eks-anywhere-build-tooling/pull/2821 for background and path forward
+RUN echo "force runc version ... " \
+    && DEBIAN_FRONTEND=noninteractive clean-install runc-1.1.5-1.amzn2023.0.1
+
 # NOTE: systemd-binfmt.service will register things into binfmt_misc which is kernel-global
 RUN echo "Enabling / Disabling services ... " \
     && systemctl enable kubelet.service \
-- 
2.48.1

