From 04ce891cae930cc9c1b08ff68aeeb727166d6376 Mon Sep 17 00:00:00 2001
From: Prow Bot <prow@amazonaws.com>
Date: Sat, 12 Aug 2023 12:08:58 -0500
Subject: [PATCH] TEMP: lock containerd and runc version

---
 images/base/Dockerfile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/images/base/Dockerfile b/images/base/Dockerfile
index e4c13a47..f38be515 100644
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -103,6 +103,9 @@ RUN echo "Installing Packages ..." \
 RUN echo "Enabling kubelet ... " \
     && systemctl enable kubelet.service
 
+RUN echo "force runc and containerd version ... " \
+    && DEBIAN_FRONTEND=noninteractive clean-install containerd-1.6.19-1.amzn2023.0.1 runc-1.1.5-1.amzn2023.0.1
+
 RUN echo "Enabling containerd ..." \
     && ctr oci spec \
         | jq '.hooks.createContainer[.hooks.createContainer| length] |= . + {"path": "/usr/local/bin/mount-product-files"}' \
-- 
2.40.1

