From c292cc942c75c75072deb129f670df26b6697c45 Mon Sep 17 00:00:00 2001
From: Jackson West <jgw@amazon.com>
Date: Sun, 20 Aug 2023 14:30:06 -0500
Subject: [PATCH 5/5] TEMP: lock containerd and runc version

---
 images/base/Dockerfile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/images/base/Dockerfile b/images/base/Dockerfile
index 68e07d1e..c91b5430 100644
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -98,6 +98,8 @@ RUN echo "Installing Packages ..." \
     # leaving for now, but al23 may not be affected by this issue
     && systemctl mask getty@tty1.service
     
+RUN echo "force runc and containerd version ... " \
+    && DEBIAN_FRONTEND=noninteractive clean-install runc-1.1.5-1.amzn2023.0.1
 
 RUN echo "Enabling services ... " \
     && systemctl enable kubelet.service \
-- 
2.34.1

