From 49fbf65b3cfa2cc709e1e30228e50b31003f2f3a Mon Sep 17 00:00:00 2001
From: Jackson West <jgw@amazon.com>
Date: Sun, 20 Aug 2023 14:30:06 -0500
Subject: [PATCH 5/8] TEMP: lock containerd and runc version

---
 images/base/Dockerfile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/images/base/Dockerfile b/images/base/Dockerfile
index b3b7e848..2f67afd7 100644
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -100,6 +100,10 @@ RUN chmod 755 /kind/bin && \
     # leaving for now, but al23 may not be affected by this issue
     && systemctl mask getty@tty1.service
 
+# see https://github.com/aws/eks-anywhere-build-tooling/pull/2821 for background and path forward
+RUN echo "force runc version ... " \
+    && DEBIAN_FRONTEND=noninteractive clean-install runc-1.1.5-1.amzn2023.0.1
+
 # NOTE: systemd-binfmt.service will register things into binfmt_misc which is kernel-global
 RUN echo "Enabling / Disabling services ... " \
     && systemctl enable kubelet.service \
-- 
2.50.1 (Apple Git-155)

