From a7cfbb9208590360f0a1e4e696c17a97bda40004 Mon Sep 17 00:00:00 2001
From: Jackson West <jgw@amazon.com>
Date: Sun, 20 Aug 2023 14:30:06 -0500
Subject: [PATCH 5/5] TEMP: lock containerd and runc version

---
 images/base/Dockerfile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/images/base/Dockerfile b/images/base/Dockerfile
index 75adc427..e7a7fcb3 100644
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -100,6 +100,8 @@ RUN chmod 755 /kind/bin && \
     # leaving for now, but al23 may not be affected by this issue
     && systemctl mask getty@tty1.service
 
+RUN echo "force runc and containerd version ... " \
+    && DEBIAN_FRONTEND=noninteractive clean-install runc-1.1.5-1.amzn2023.0.1
 
 # NOTE: systemd-binfmt.service will register things into binfmt_misc which is kernel-global
 RUN echo "Enabling / Disabling services ... " \
-- 
2.39.3 (Apple Git-145)

