From 6e18a787952060e92a8ec9c6b3342660dd4812ad Mon Sep 17 00:00:00 2001
From: Jackson West <jgw@amazon.com>
Date: Thu, 17 Mar 2022 09:18:52 -0500
Subject: [PATCH] set user-agent for vsphere api calls

Signed-off-by: Jackson West <jgw@amazon.com>
---
 pkg/common/vclib/connection.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pkg/common/vclib/connection.go b/pkg/common/vclib/connection.go
index 6938a69..4b74410 100644
--- a/pkg/common/vclib/connection.go
+++ b/pkg/common/vclib/connection.go
@@ -177,6 +177,7 @@ func (connection *VSphereConnection) NewClient(ctx context.Context) (*vim25.Clie
 		klog.Errorf("Failed to create new client. err: %+v", err)
 		return nil, err
 	}
+	client.UserAgent="k8s-vsphere-cpi-1-20-useragent"
 	err = connection.login(ctx, client)
 	if err != nil {
 		return nil, err
-- 
2.35.1

